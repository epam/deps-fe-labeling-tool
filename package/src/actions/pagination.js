import { createAction } from 'redux-actions'

const FEATURE_NAME = 'PAGINATION'

const openPage = createAction(`${FEATURE_NAME}/OPEN_PAGE`)

const resetDefault = createAction(
  `${FEATURE_NAME}/RESET_DEFAULT`
)

export {
  resetDefault,
  openPage
}
