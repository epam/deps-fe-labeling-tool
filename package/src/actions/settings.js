import { createAction } from 'redux-actions'

const FEATURE_NAME = 'SETTINGS'

const storeSettings = createAction(`${FEATURE_NAME}/STORE`)

const resetDefault = createAction(`${FEATURE_NAME}/RESET_DEFAULT`)

export {
  resetDefault,
  storeSettings
}
