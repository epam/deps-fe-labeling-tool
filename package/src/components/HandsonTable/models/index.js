
export * from './HTTableData'
export * from './HTMerge'
export * from './HTColumn'
