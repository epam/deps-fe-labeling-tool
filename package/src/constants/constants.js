const CONFIDENCE_ON_MANUAL_CHANGE = 1

export {
  CONFIDENCE_ON_MANUAL_CHANGE
}
