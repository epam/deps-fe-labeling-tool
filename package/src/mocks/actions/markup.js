const addLabels = jest.fn((labels) => ({
  type: 'MOCK_ADD_LABELS_ACTION',
  payload: labels
}))

const addTables = jest.fn((tables) => ({
  type: 'MOCK_ADD_TABLES_ACTION',
  payload: tables
}))

const removeLabels = jest.fn((labels) => ({
  type: 'MOCK_REMOVE_LABELS_ACTION',
  payload: labels
}))

const updateLabels = jest.fn((labels) => ({
  type: 'MOCK_UPDATE_LABELS_ACTION',
  payload: labels
}))

const updateAllLabels = jest.fn((labels) => ({
  type: 'MOCK_UPDATE_ALL_LABELS_ACTION',
  payload: labels
}))

const updateTables = jest.fn((tables) => ({
  type: 'MOCK_UPDATE_TABLES_ACTION',
  payload: tables
}))

const updateAllTables = jest.fn((tables) => ({
  type: 'MOCK_UPDATE_ALL_TABLES_ACTION',
  payload: tables
}))

const selectLabels = jest.fn((labels) => ({
  type: 'MOCK_SELECT_LABELS_ACTION',
  payload: labels
}))

const selectTables = jest.fn((tables) => ({
  type: 'MOCK_SELECT_TABLES_ACTION',
  payload: tables
}))

const clearSelection = jest.fn(() => ({
  type: 'MOCK_CLEAR_SELECTION_ACTION'
}))

const removeTables = jest.fn((tables) => ({
  type: 'MOCK_REMOVE_TABLES_ACTION',
  payload: tables
}))

const importMarkup = jest.fn(() => ({
  type: 'MOCK_IMPORT_MARKUP'
}))

const exportMarkup = jest.fn(() => ({
  type: 'MOCK_EXPORT_MARKUP'
}))

const undo = jest.fn(() => ({
  type: 'MOCK_UNDO_ACTION'
}))

const redo = jest.fn(() => ({
  type: 'MOCK_REDO_ACTION'
}))

const updateInitialMarkup = jest.fn(() => ({
  type: 'MOCK_UPDATE_INITIAL_MARKUP_ACTION'
}))

const storeMarkup = jest.fn((markup) => ({
  type: 'MOCK_STORE_MARKUP_ACTION',
  payload: markup
}))

const storeAssignedMarkup = jest.fn((markup) => ({
  type: 'MOCK_UPDATE_MARKUP_ACTION',
  payload: markup
}))

const updatePageMarkup = jest.fn(() => ({
  type: 'UPDATE_PAGE_MARKUP'
}))

const reset = jest.fn(() => ({
  type: 'RESET_MARKUP_ACTION'
}))

const copyMarkup = jest.fn(() => ({
  type: 'COPY_MARKUP_ACTION'
}))

const pasteMarkup = jest.fn(() => ({
  type: 'PASTE_MARKUP_ACTION'
}))

const updateLabelsWithSettings = jest.fn(() => ({
  type: 'UPDATE_LABELS_WITH_SETTINGS'
}))

const updateTablesWithSettings = jest.fn(() => ({
  type: 'UPDATE_TABLES_WITH_SETTINGS'
}))

const mockMarkupActions = {
  addLabels,
  addTables,
  copyMarkup,
  pasteMarkup,
  removeLabels,
  reset,
  updateLabels,
  updateAllLabels,
  updateTables,
  updateAllTables,
  selectLabels,
  selectTables,
  clearSelection,
  removeTables,
  importMarkup,
  exportMarkup,
  undo,
  updateInitialMarkup,
  storeMarkup,
  storeAssignedMarkup,
  updatePageMarkup,
  redo,
  updateLabelsWithSettings,
  updateTablesWithSettings
}

export {
  mockMarkupActions
}
