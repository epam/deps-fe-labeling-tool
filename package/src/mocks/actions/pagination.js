const openPage = jest.fn((page) => ({
  type: 'MOCK_OPEN_PAGE_ACTION',
  payload: page
}))

const mockPaginationActions = {
  openPage
}

export {
  mockPaginationActions
}
