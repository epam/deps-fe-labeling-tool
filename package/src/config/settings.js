import { Settings } from '@/models/Settings'

const settings = new Settings()

export {
  settings
}
