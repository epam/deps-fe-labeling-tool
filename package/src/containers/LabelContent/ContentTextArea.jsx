import React, { useMemo, useCallback } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { updateLabels } from '@/actions/markup'
import { CONFIDENCE_ON_MANUAL_CHANGE } from '@/constants/constants'
import { SpecialSymbols } from '@/containers/SpecialSymbols'
import { Feature } from '@/enums/Feature'
import { debounce } from '@/hocs/debounce'
import { labelShape } from '@/models/Label'
import { Settings, settingsShape } from '@/models/Settings'
import { currentPageSelector } from '@/selectors/pagination'
import { settingsSelector } from '@/selectors/settings'
import { TextAreaStyled } from './ContentTextArea.styles'

const DebouncedTextArea = debounce({
  getValue: (e) => e.target.value
})(TextAreaStyled)

const ContentTextArea = ({
  currentPage,
  label,
  settings,
  updateLabels
}) => {
  const areSymbolsAvailable = useMemo(
    () => (
      Settings.has(settings, Feature.SPECIAL_SYMBOLS)
    ),
    [settings]
  )

  const onContentChange = useCallback(
    (value) => {
      updateLabels(currentPage, [{
        ...label,
        content: value,
        confidence: CONFIDENCE_ON_MANUAL_CHANGE
      }])
    },
    [
      currentPage,
      label,
      updateLabels
    ])

  return (
    <>
      <DebouncedTextArea
        rows={2}
        value={label.content}
        onChange={onContentChange}
      />
      {
        areSymbolsAvailable && (
          <SpecialSymbols
            label={label}
          />
        )
      }
    </>
  )
}

ContentTextArea.propTypes = {
  currentPage: PropTypes.number.isRequired,
  label: labelShape.isRequired,
  settings: settingsShape.isRequired,
  updateLabels: PropTypes.func.isRequired
}

const mapStateToProps = (state) => ({
  currentPage: currentPageSelector(state),
  settings: settingsSelector(state)
})

const mapDispatchToProps = {
  updateLabels
}

const ConnectedComponent = connect(mapStateToProps, mapDispatchToProps)(ContentTextArea)

export {
  ConnectedComponent as ContentTextArea
}
