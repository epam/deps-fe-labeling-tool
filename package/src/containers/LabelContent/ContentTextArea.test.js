import { mockMarkupActions } from '@/mocks/actions/markup'
import { mockComponent } from '@/mocks/mockComponent'
import { mockReactRedux } from '@/mocks/mockReactRedux'
import { mockUuid } from '@/mocks/mockUuid'
import { mockPaginationSelectors } from '@/mocks/selectors/pagination'
import { mockSettingsSelectors } from '@/mocks/selectors/settings'
import React from 'react'
import { shallow } from 'enzyme'
import { updateLabels } from '@/actions/markup'
import { CONFIDENCE_ON_MANUAL_CHANGE } from '@/constants/constants'
import { Label, LabelType } from '@/models/Label'
import { ContentTextArea } from './ContentTextArea'
import { TextAreaStyled } from './ContentTextArea.styles'

jest.mock('react-redux', () => mockReactRedux)
jest.mock('@/actions/markup', () => mockMarkupActions)
jest.mock('@/selectors/pagination', () => mockPaginationSelectors)
jest.mock('@/selectors/settings', () => mockSettingsSelectors)
jest.mock('@/containers/SpecialSymbols', () =>
  mockComponent('SpecialSymbols')
)
jest.mock('@/hocs/debounce', () => ({
  debounce: () => (Component) => Component
}))
jest.mock('uuid', () => mockUuid)

const {
  WrappedComponent,
  mapStateToProps,
  mapDispatchToProps
} = ContentTextArea

describe('Container: ContentTextArea', () => {
  describe('mapStateToProps', () => {
    it('should pass expected state as props to WrappedComponent', () => {
      const { props } = mapStateToProps()
      const expectedState = {
        currentPage: mockPaginationSelectors.currentPageSelector(),
        settings: mockSettingsSelectors.settingsSelector()
      }
      expect(props).toEqual(expectedState)
    })
  })

  describe('mapDispatchToProps', () => {
    const { props } = mapDispatchToProps()
    it('should pass updateLabels action as prop to WrappedComponent', () => {
      props.updateLabels()
      expect(updateLabels).toHaveBeenCalled()
    })
  })

  describe('component', () => {
    let wrapper, defaultProps

    beforeEach(() => {
      defaultProps = {
        ...mapStateToProps().props,
        ...mapDispatchToProps().props,
        label: new Label(
          0.113,
          0.8104,
          0.017,
          0.0119,
          'enclosures',
          undefined,
          LabelType.CHECKMARK,
          true,
          { data: 'some meta for enclosures value label' },
          0.66
        )
      }
      wrapper = shallow(<WrappedComponent {...defaultProps} />)
    })

    it('should render correct layout based on props', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should call updateLabels prop with correct args when calling TextArea onChange (onContentChange)', () => {
      const TextAreaWrapper = wrapper.find(TextAreaStyled)
      const mockValue = 'mockValue'
      TextAreaWrapper.props().onChange(mockValue)

      const expectedSecondArg = [
        {
          ...defaultProps.label,
          content: mockValue,
          confidence: CONFIDENCE_ON_MANUAL_CHANGE
        }
      ]

      expect(updateLabels).nthCalledWith(
        1,
        defaultProps.currentPage,
        expectedSecondArg
      )
    })
  })
})
