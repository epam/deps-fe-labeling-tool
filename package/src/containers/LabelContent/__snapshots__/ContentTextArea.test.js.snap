// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: ContentTextArea component should render correct layout based on props 1`] = `
<Fragment>
  <ContentTextAreastyles__TextAreaStyled
    onChange={[Function]}
    rows={2}
    value={true}
  />
  <SpecialSymbols
    label={
      Label {
        "confidence": 0.66,
        "content": true,
        "fieldCode": "enclosures",
        "h": 0.0119,
        "index": undefined,
        "meta": Object {
          "data": "some meta for enclosures value label",
        },
        "type": "checkmark",
        "typeName": "label",
        "uid": "1",
        "w": 0.017,
        "x": 0.113,
        "y": 0.8104,
      }
    }
  />
</Fragment>
`;
